File: CSharp Version/EmergencyResponseSystem_DAL/Services/PatientService.cs
using Dapper;
using EmergencyResponseSystem.Models;
using EmergencyResponseSystem_DAL.Database;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EmergencyResponseSystem_DAL.Services
{
    public class PatientService
    {
        public async Task<List<Patient>> GetAllAsync()
        {
            using (var connection = DBManager.GetDBConnexion(true))
            {
                var query = "SELECT Id, Nom, Prenom, NomJF, DateNaissance, Adresse, Ville, CodePostal, NumINSEE, Location, SpecialiteRequise FROM Patient";
                var result = await connection.QueryAsync<Patient>(query);
                return result.ToList();
            }
        }

        public async Task<Patient> GetAsyncByNumINSEE(string numINSEE)
        {
            using (var connection = DBManager.GetDBConnexion(true))
            {
                var query = "SELECT Id, Nom, Prenom, NomJF, DateNaissance, Adresse, Ville, CodePostal, NumINSEE, Location, SpecialiteRequise FROM Patient WHERE NumINSEE = @NumINSEE";
                var result = await connection.QuerySingleOrDefaultAsync<Patient>(query, new { NumINSEE = numINSEE });
                return result;
            }
        }
        public async Task<Patient> GetAsync(int id)
        {
            using (var connection = DBManager.GetDBConnexion(true))
            {
                var query = "SELECT Id, Nom, Prenom, NomJF, DateNaissance, Adresse, Ville, CodePostal, NumINSEE, Location, SpecialiteRequise FROM Patient WHERE Id = @Id";
                var result = await connection.QuerySingleOrDefaultAsync<Patient>(query, new { Id = id });
                return result;
            }
        }

        public async Task<string> InsertAsync(Patient patient)
        {
            using (var connection = DBManager.GetDBConnexion(true))
            {
                var query = "INSERT INTO Patient (Nom, Prenom, NomJF, DateNaissance, Adresse, Ville, CodePostal, NumINSEE, Location, SpecialiteRequise) VALUES (@Nom, @Prenom, @NomJF, @DateNaissance, @Adresse, @Ville, @CodePostal, @NumINSEE, @Location, @SpecialiteRequise)";
                await connection.ExecuteAsync(query, patient);
                return patient.NumINSEE;
            }
        }

        public async Task UpdateAsync(Patient patient)
        {
            using (var connection = DBManager.GetDBConnexion(true))
            {
                var query = "UPDATE Patient SET Nom = @Nom, Prenom = @Prenom, NomJF = @NomJF, DateNaissance = @DateNaissance, Adresse = @Adresse, Ville = @Ville, CodePostal = @CodePostal, Location = @Location, SpecialiteRequise = @SpecialiteRequise WHERE NumINSEE = @NumINSEE";
                await connection.ExecuteAsync(query, patient);
            }
        }

        public async Task DeleteAsyncByNumINSEE(string numINSEE)
        {
            using (var connection = DBManager.GetDBConnexion(true))
            {
                var query = "DELETE FROM Patient WHERE NumINSEE = @NumINSEE";
                await connection.ExecuteAsync(query, new { NumINSEE = numINSEE });
            }
        }
        public async Task DeleteAsync(int id)
        {
            using (var connection = DBManager.GetDBConnexion(true))
            {
                var query = "DELETE FROM Patient WHERE Id = @Id";
                await connection.ExecuteAsync(query, new { Id = id });
            }
        }
    }
}

File: Backend/src/main/java/eu/echodream/medhead/Services/PatientService.java
package eu.echodream.medhead.Services;

import eu.echodream.medhead.Models.Patient;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.jdbc.core.RowMapper;
import org.springframework.stereotype.Service;

import java.sql.Date;
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.util.List;

@Service
public class PatientService {
    private final JdbcTemplate jdbcTemplate;

    public PatientService(JdbcTemplate jdbcTemplate) {
        this.jdbcTemplate = jdbcTemplate;
    }

    public List<Patient> getAll() {
        String query = "SELECT Id, Nom, Prenom, NomJF, DateNaissance, Adresse, Ville, CodePostal, NumINSEE, HopitalActuel, SpecialiteRequise FROM Patient";
        return jdbcTemplate.query(query, rowMapper);
    }

    public Patient getByNumINSEE(String numINSEE) {
        String query = "SELECT Id, Nom, Prenom, NomJF, DateNaissance, Adresse, Ville, CodePostal, NumINSEE, HopitalActuel, SpecialiteRequise FROM Patient WHERE NumINSEE = ?";
        return jdbcTemplate.queryForObject(query, new Object[]{numINSEE}, rowMapper);
    }

    public Patient get(int id) {
        String query = "SELECT Id, Nom, Prenom, NomJF, DateNaissance, Adresse, Ville, CodePostal, NumINSEE, HopitalActuel, SpecialiteRequise FROM Patient WHERE Id = ?";
        return jdbcTemplate.queryForObject(query, new Object[]{id}, rowMapper);
    }

    public int insert(Patient patient) {
        String query = "INSERT INTO Patient (Nom, Prenom, NomJF, DateNaissance, Adresse, Ville, CodePostal, NumINSEE, HopitalActuel, SpecialiteRequise) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?);";
        jdbcTemplate.update(query, patient.getNom(), patient.getPrenom(), patient.getNomJF(), patient.getDateNaissance().toString(), patient.getAdresse(), patient.getVille(), patient.getCodePostal(), patient.getNumINSEE(), patient.getHopitalActuel(), patient.getSpecialiteRequise());
        return jdbcTemplate.queryForObject("SELECT MAX(Id) FROM Patient", Integer.class);
    }


    public void update(Patient patient) {
        String query = "UPDATE Patient SET Nom = ?, Prenom = ?, NomJF = ?, DateNaissance = ?, Adresse = ?, Ville = ?, CodePostal = ?, HopitalActuel = ?, SpecialiteRequise = ? WHERE NumINSEE = ?";
        jdbcTemplate.update(query, patient.getNom(), patient.getPrenom(), patient.getNomJF(),
                patient.getDateNaissance(), patient.getAdresse(), patient.getVille(),
                patient.getCodePostal(), patient.getHopitalActuel(), patient.getSpecialiteRequise(), patient.getNumINSEE());
    }

    public void deleteByNumINSEE(String numINSEE) {
        String query = "DELETE FROM Patient WHERE NumINSEE = ?";
        jdbcTemplate.update(query, numINSEE);
    }

    public void delete(int id) {
        String query = "DELETE FROM Patient WHERE Id = ?";
        jdbcTemplate.update(query, id);
    }

    private static final RowMapper<Patient> rowMapper = (rs, rowNum) -> {
        int id = rs.getInt("Id");
        String nom = rs.getString("Nom");
        String prenom = rs.getString("Prenom");
        String nomJF = rs.getString("NomJF");
        String adresse = rs.getString("Adresse");
        String ville = rs.getString("Ville");
        String codePostal = rs.getString("CodePostal");
        String numINSEE = rs.getString("NumINSEE");
        int hopitalActuel = rs.getInt("HopitalActuel");
        int specialiteRequise = rs.getInt("SpecialiteRequise");

        // Supposons que la date est stockée en tant que String dans le format "yyyy-MM-dd"
        String dateString = rs.getString("DateNaissance");
        LocalDate dateNaissance = LocalDate.parse(dateString, DateTimeFormatter.ISO_LOCAL_DATE);

        return new Patient(id, nom, prenom, nomJF, Date.valueOf(dateNaissance), adresse, ville, codePostal, numINSEE, hopitalActuel, specialiteRequise);
        };
    }